### PR TITLE
CoreTiming: Pass Core::System to Events.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
More steps towards de-globalization.

### CODE REVIEW

File: Source/Core/Core/HW/EXI/EXI.cpp [hunk 1]
```diff
@@ -270,7 +270,7 @@ void UpdateInterrupts()
   ProcessorInterface::SetInterrupt(ProcessorInterface::INT_CAUSE_EXI, causeInt);
 }
 
-static void UpdateInterruptsCallback(u64 userdata, s64 cycles_late)
+static void UpdateInterruptsCallback(Core::System& system, u64 userdata, s64 cycles_late)
```
Comment (iwubcode) [inline Source/Core/Core/HW/EXI/EXI.cpp:273]: Just curious but what was this added for?

Comment (JosJuice) [reply Source/Core/Core/HW/EXI/EXI.cpp:273]: It won't compile unless the function signature of this callback matches.

Comment (AdmiralCurtiss) [reply Source/Core/Core/HW/EXI/EXI.cpp:273]: Can't pass a function pointer unless the signature matches.

Comment (iwubcode) [reply Source/Core/Core/HW/EXI/EXI.cpp:273]: Thank you both, hadn't gotten that far in my investigation yet. That makes sense!

**Review by lioncash — APPROVED** [review]

**Pull request merged**